### PR TITLE
Star challenge: Nicer binned files for PZ to use

### PR DESCRIPTION
**Pull request by joezuntz**
This adds a pipeline to generate nicer files for the PZ group to start testing with, and some supporting code.

- A script to make a metacal file into a metadetect one, so I can reuse some old files instead of re-ingesting
- The pipeline YML files
- Moves the random forest binning so it can be used by both source and lens
- Adds a lensing tomography code using the random forest
- Allows saving extra columns in the stages that split the catalogs into bins

### CODE REVIEW

File: txpipe/binning/random_forest.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    # classification, to ensure that everything is consistent. In that
+    # case if we are not the root process, wait for them to finish and
+    # receive and return their classifier
+    if (comm is not None) and (comm.rank > 0):
+        classifier = comm.bcast(None)
+        features = comm.bcast(None)
+        return classifier, features
+
+    # Load the training data
+    training_data_table = Table.read(training_file, format="ascii")
```
[the remark this answers is not in the document]
Comment (joezuntz) [reply txpipe/binning/random_forest.py:20]: Yes, definitely. This training data is fully complete, even at high-z IIRC. It's a placeholder for running with more sophisticated training processes with RAIL.

Although this is for the tomography not for the n(z) or p(z), so it's not necessarily clear what we want to do in real life here - if our training table is incomplete we might want to augment it with best-guess mock values.

[the remark this answers is not in the document]
Comment (chihway) [reply txpipe/binning/random_forest.py:20]: yep makes sense

**Comment by chihway** [comment]
This all looks good -- feel free to merge. 

Just a note -- It is currently quite a different sample than project #228 (both lens and source), though we might find out later that this is actually better if we can't fix the redmagic covariance, and our decision of the 7-bin source sample is pretty arbitrary anyway.

**Comment by joezuntz** [comment]
Thanks!

**Pull request merged**